compiler/cpp/src/thrift/generate/t_java_generator.cc

```diff
@@ -5265,6 +5274,9 @@ THRIFT_REGISTER_GENERATOR(
     "    android_legacy:  Do not use java.io.IOException(throwable) (available for Android 2.3 and "
     "above).\n"
     "    option_type:     Wrap optional fields in an Option type.\n"
+    "    handle_runtime_exceptions:\n"
+    "                     Generated services will handle RuntimeException as "
```

Comment:
I don't think we need to mention anything about the client side w.r.t to TTransportException.  Saying anything about the client side is hard because of all the different language implementations.  

However this code is focused on generating a java server, so we can say what the java server will do.  The java server will either send TApplicationException or close the connection depending on this setting.